src/tof/result.py

```diff
@@ -313,13 +315,11 @@ def plot(
             self._plot_pulse(pulse_index=i, ax=ax)
 
         det_data = furthest_detector.tofs.visible.data
-        if len(det_data) == 1:
-            tof_max = det_data['pulse:0'].coords['tof'].max().value
+        if sum([da.sum().value for da in det_data.values()]) > 0:
+            times = [da.coords['tof'].max() for da in det_data.values()]
```

Comment:
```suggestion
        if sum(da.sum().value for da in det_data.values()) > 0:
            times = (da.coords['tof'].max() for da in det_data.values())
```
and the else branch.